### PR TITLE
Onboarding API Management Module (Preview Support)

### PR DESCRIPTION
**Pull request by kevinhillinger**
Ref Issue #3614

Myself and @jonlester have implemented the start of the module for API Management. Typically, we would communicate our desire to contribute early and create consensus with the CLI maintainers, but given the nature of the program (one week hack) for this week, we ask for forgiveness on this :)

We believe we have adhered to all development guidelines in developing the module for API Management, including tests and including the recording. Here is the summary of the feature that is included:

- new module for APIM named "apim"
- Initial support for service commands: list, show, create, update, delete

### CODE REVIEW

**Comment by kevinhillinger** [comment]
@zikalino Small world! I'm looking forward to getting this work reviewed and get API Management incrementally across the finish line. Thanks!

**Comment by kevinhillinger** [comment]
@yugangw-msft @zikalino @tjprescott 

We'd love to continue to support the addition subcommand groups for this module, but need an accepted base for apim.

File: src/azure-cli/azure/cli/command_modules/apim/_help.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+short-summary: Deletes an API Management service.
+examples:
+  - name: Delete an API Management service.
+    text: >
+        az apim delete -n MyApim -g MyResourceGroup
+"""
+
+helps['apim show'] = """
+    type: command
+    short-summary: Show details of an APIM instance.
```
Comment (zikalino) [inline src/azure-cli/azure/cli/command_modules/apim/_help.py]: minor, should be "API Management service instance"

File: src/azure-cli/azure/cli/command_modules/apim/_help.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+"""
+
+helps['apim update'] = """
+    type: command
+    short-summary: Update an API Management serviceaz  instance.
```
Comment (zikalino) [inline src/azure-cli/azure/cli/command_modules/apim/_help.py]: minor: serviceaz  instance -> service instance

File: src/azure-cli/azure/cli/command_modules/apim/_help.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+"""
+
+helps['apim backup'] = """
+    type: command
+    short-summary: Creates a backup of the API Management service to the given Azure
+        Storage Account. This is long running operation and could take several
+        minutes to complete.az
```
Comment (zikalino) [inline src/azure-cli/azure/cli/command_modules/apim/_help.py]: remove "az" at the end

File: src/azure-cli/azure/cli/command_modules/apim/_client_factory.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.apimanagement import ApiManagementClient
+    return get_mgmt_service_client(cli_ctx, ApiManagementClient)
+
+
+def cf_service(cli_ctx, *_):
+    return cf_apim(cli_ctx).api_management_service
+
+
+def cf_api(cli_ctx, *_):
```
Comment (zikalino) [inline src/azure-cli/azure/cli/command_modules/apim/_client_factory.py]: seems this is not used yet

**Pull request merged**